### ISSUE TITLE
.foreign & .withSchema ?

### ISSUE THREAD

**Issue by ash0080** [knex/knex#4910]
# Environment

Knex version:  0.95.15
Database + version: 
OS:
PostgreSQL 12.7 (Debian 12.7-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do
```
k.schema
	.withSchema('test1')
	.createTable('u_test', t => {
		t.increments('id').primary()
		t.foreign('test_id').references('id').inTable('test')
			.onDelete('CASCADE')
			.onUpdate('NO ACTION')
		t.timestamps()
	})
```
when I use .withSchema(), of course I want the foreign key to refer to test.id under the **same** schema, however, 42P01 appears here because the actual references are default **public** schema

although I could fix it like this,
```.inTable('test1.test')```
But it doesn't seem to be compliant
I think it would be more syntactic to either modify the presentation of .withSchema() or to provide a .foreign() sibling modifier.

2. Error message
42P01

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

**Comment by rajivkr** [comment]
Facing the same issue, any updates on this ? it would be great if we the callback to the `createTable` function picks the schema from the `withSchema` method

**Comment by ir33k** [comment]
Agree, I think that `foreign()` should use schema from `createTable()`.  Also it should be possible to alter schema name using `withSchema()` on `foreign()` if you need to reference table from different schema (which is the problem I'm facing).

**Comment by mercmobily** [comment]
Thank you for reporting this issue. This is a duplicate of #1614, which has already been triaged and labeled as an NG_bug.

Both issues describe the same problem: when using `withSchema()` with foreign key references, the schema prefix is not automatically applied to the referenced table, causing PostgreSQL to look for the table in the default (public) schema instead of the specified schema.

The issue will be resolved by PR #6169, which has been updated to include automatic schema inheritance. When you use `withSchema('test1')` and create a foreign key reference like `table.foreign('test_id').references('id').inTable('test')`, the schema context will now be automatically applied, correctly generating a reference to `test1.test.id` rather than `public.test.id`.

The PR also fixes the related parsing bug where explicitly specifying a schema-qualified reference (like `testSchema.LeftSide.LeftSideId`) was being incorrectly parsed, dropping the column name entirely.

Your workaround using `inTable('test1.test')` will continue to work after the fix is merged, ensuring backward compatibility.

Related Issues:
- #1614 - Using `references` when creating a table in a named schema results in error (OPEN, NG_bug - master issue)
- #4910 - .foreign & .withSchema? (OPEN - duplicate)
- #3115 - Migration failing with error relation does not exist (OPEN)
- PR #6169 - Fix for foreign key's referencing table schemas (OPEN)
- PR #5651 - Fix .foreign to utilize withSchema setting (CLOSED)
